File: seqkit/fq2fa.go
package seqkit

import (
	"ignis/driver/api"
)

type SeqKitFq2FaOptions struct {
	inner Fq2FaOptions
}

type Fq2FaOptions struct {
	Config KitConfig
}

func (this *Fq2FaOptions) setDefaults() *Fq2FaOptions {
	this.Config.setDefaults()
	return this
}

func (this *SeqKitFq2FaOptions) Config(v *SeqKitConfig) *SeqKitFq2FaOptions {
	this.inner.Config = v.inner
	return this
}

func Fq2Fa(input *api.IDataFrame[string], o *SeqKitFq2FaOptions) (*api.IDataFrame[string], error) {
	if o == nil {
		o = &SeqKitFq2FaOptions{}
	}
	opts := o.inner
	opts.setDefaults()
	libprepare, err := api.AddParam(libSource("Fq2Fa"), "opts", OptionsToString(o.inner))
	if err != nil {
		return nil, err
	}

	return api.MapPartitions[string, string](input, libprepare)
}

File: bigseqkit-py/bigseqkit/fq2fa.py
from bigseqkit.helper import _setDefault, _libSource, _config, _optionsToString, _parseKargs, SeqKitConfig, IDataFrame


class SeqKitFq2FaOptions:

    def __init__(self):
        self.__inner = Fq2FaOptions()

    def config(self, v: SeqKitConfig):
        self.__inner.Config = _config(v)

    def _run(self, input: IDataFrame, **kwargs):
        opts = self.__inner
        _parseKargs(opts, kwargs)
        opts.setDefaults()
        libprepare = _libSource("Fq2Fa").addParam("opts", _optionsToString(opts))
        return input.mapPartitions(libprepare)


class Fq2FaOptions:

    def __init__(self):
        self.Config = None  # KitConfig

    def setDefaults(self):
        _setDefault(self, "Config", _config(SeqKitConfig())).setDefaults()


def fq2fa(input: IDataFrame, o: SeqKitFq2FaOptions = None, **kwargs):
    if o is None:
        o = SeqKitFq2FaOptions()
    return o._run(input, **kwargs)
